include/my_sys.h

```diff
@@ -921,16 +921,31 @@ extern int my_compress_buffer(uchar *dest, size_t *destLen,
 extern int packfrm(const uchar *, size_t, uchar **, size_t *);
 extern int unpackfrm(uchar **, size_t *, const uchar *);
 
-extern ha_checksum my_checksum(ha_checksum crc, const uchar *mem,
-                               size_t count);
+void my_checksum_init(void);
+#ifdef HAVE_CRC32_VPMSUM
+extern my_checksum(ha_checksum, const void *, size_t);
+#else
+typedef ha_checksum (*my_crc32_t)(ha_checksum, const void *, size_t);
+extern my_crc32_t my_checksum;
+#endif
```

Comment:
I like the fact that POWER with its compile-time detection of the `vpmsum` instruction avoids the function pointer overhead altogether.

Reply:
right. Unfortunately, other platforms have to evaluate runtime decisions based on the capabilities but I am not sure why function-pointers are bad. Do they incur the visible cost?

Reply:
The cost should be minimal. Until now, there was a similar cost for invoking `crc32()` when zlib was linked dynamically. (And with this change, the non-accelerated `crc32()` would incur a cost of two function pointer calls. I think that it is still acceptable.)